### PR TITLE
feat(llmobs): modify LLMObs.annotate() to meet new span schema

### PR DESCRIPTION
**Pull request by Yun-Kim**
This PR does a couple things to the `LLMObs.annotate()` method:
- All non-LLM kind spans will cast input/output.value set as text, no longer as messages
- Adds `ddtrace.llmobs.utils.Message/Messages` helper classes to allow users to format messages into a standard type.
- Deprecate setting parameters (replaced by setting tags instead)
- Add error handling and logging instead of crashing during annotation due to bad types
- Update tags rather than override
- Improve docstrings
- Update langchain llmobs code to try serializing chain input/outputs into JSON

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/llmobs/_llmobs.py [hunk 1]
```diff
@@ -272,78 +276,104 @@ def annotate(
...
-        else:
-            if isinstance(data, str):
-                if io_type == "input":
-                    span.set_tag_str(INPUT_VALUE, data)
-                else:
-                    span.set_tag_str(OUTPUT_VALUE, data)
-            else:
-                log.warning("Invalid input/output type for non-llm span. Must be a raw string.")
+        if input_value is not None:
+            span.set_tag_str(INPUT_VALUE, str(input_value))
```
Comment (lievan) [inline ddtrace/llmobs/_llmobs.py:339]: I think the frontend only picks up the JSON rendering if we do
`json.dumps(input_value)`

So we might need to do a check like this, something like
```
if isinstance(input_value, str):
      span.set_tag_str(INPUT_VALUE, input_value)
else:
     span.set_tag_str(INPUT_VALUE, json.dumps(input_value))
```

File: ddtrace/llmobs/_integrations/langchain.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def llmobs_set_tags(
         span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
         span.set_tag_str(MODEL_PROVIDER, model_provider or "")
 
-        self._llmobs_set_input_parameters(span, model_provider)
+        self._llmobs_set_input_metadata(span, model_provider)
```
Comment (lievan) [inline ddtrace/llmobs/_integrations/langchain.py:55]: Might be worth renaming this to 

`self._llmobs_set_span_metadata`

Comment (lievan) [reply ddtrace/llmobs/_integrations/langchain.py:55]: To clarify it's metadata for the whole span, not just the input

Comment (Yun-Kim) [reply ddtrace/llmobs/_integrations/langchain.py:55]: Ahh good catch!

File: ddtrace/llmobs/_llmobs.py [hunk 3]
```diff
@@ -272,78 +276,116 @@ def annotate(
...
-        else:
-            if isinstance(data, str):
-                if io_type == "input":
-                    span.set_tag_str(INPUT_VALUE, data)
-                else:
-                    span.set_tag_str(OUTPUT_VALUE, data)
+                try:
+                    span.set_tag_str(INPUT_VALUE, json.dumps(input_value))
+                except TypeError:
+                    log.warning("Failed to parse input value. Input value must be serializable.")
```
Comment (lievan) [inline ddtrace/llmobs/_llmobs.py:345]: nit maybe 'JSON serializable?'

**Review by lievan — APPROVED** [review]

**Pull request merged**